src/crafting.h

```diff
@@ -24,29 +24,19 @@ typedef int nc_color; // From color.h
 using itype_id     = std::string; // From itype.h
 using requirement_id = string_id<requirement_data>;
 
-struct byproduct {
-    itype_id result;
-    int charges_mult;
-    int amount;
-
-    byproduct() : byproduct( "null" ) {}
-
-    byproduct( itype_id res, int mult = 1, int amnt = 1 )
-        : result( res ), charges_mult( mult ), amount( amnt ) {
-    }
-};
-
 struct recipe {
     private:
         std::string ident_;
 
-        friend void load_recipe( JsonObject &jsobj, const std::string &src, bool uncraft );
+        friend class recipe_dictionary;
 
     public:
-        itype_id result;
-        int time; // in movement points (100 per turn)
-        int difficulty;
-        bool valid_to_learn;
+        recipe() : skill_used( NULL_ID ) {}
+
+        itype_id result = "null";
+        int time = 0; // in movement points (100 per turn)
```

Comment:
`moves` or `time_in_moves` or `moves_to_perform` would make the comment redundant and also would eliminate confusion in the implementation.